addons/sale/wizard/payment_link_wizard.py

```diff
@@ -3,13 +3,43 @@
 
 from werkzeug import urls
 
-from odoo import api, models
-
+from odoo import api, fields, models
+from odoo.tools import float_compare
 
 class PaymentLinkWizard(models.TransientModel):
     _inherit = 'payment.link.wizard'
     _description = 'Generate Sales Payment Link'
 
+    amount_paid = fields.Monetary(
+        currency_field='currency_id',
+        help="Amount partially paid on this order.",
+        string="Already Paid."
+    )
+
+    @api.model
+    def default_get(self, fields):
+        res = super().default_get(fields)
+
+        if res['res_id'] and res['res_model'] == 'sale.order':
+            record = self.env[res['res_model']].browse(res['res_id'])
+            amount = (record.amount_total
+                           - sum(record.invoice_ids.filtered(lambda x: x.state != 'cancel')
+                                 .mapped('amount_total'))
+                           - record.amount_paid())
+            res.update({
+                'amount': amount,
+                'amount_paid': record.amount_paid(),
+                'amount_max': record.amount_total,
+            })
+
+        return res
```

Comment:
Could be handled directly in `sale.order` `_get_default_payment_link_values` method ;).

Reply:
thanks !